packages/web3-providers-ws/src/Eip1193Provider.ts

```diff
@@ -0,0 +1,112 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import {
+	EthExecutionAPI,
+	HexString,
+	Web3APIMethod,
+	Web3APIPayload,
+	Web3APISpec,
+	Web3BaseProvider,
+} from 'web3-types';
+import { EventEmitter } from 'events';
+import { toPayload } from 'web3-utils';
+
+export default abstract class Eip1193Provider<
```

Comment:
I think this abstract class should be placed in utils (or some other common) package so we can re use it in IPC provider + external users can use for making their providers eip1193